routing/route/route.go

```diff
@@ -105,19 +106,9 @@ func (h *Hop) PackHopPayload(w io.Writer, nextChanID uint64) error {
 	// required routing fields, as well as these optional values.
 	amt := uint64(h.AmtToForward)
 	combinedRecords := append(h.TLVRecords,
-		tlv.MakeDynamicRecord(
-			tlv.AmtOnionType, &amt, func() uint64 {
-				return tlv.SizeTUint64(amt)
-			},
-			tlv.ETUint64, tlv.DTUint64,
-		),
-		tlv.MakeDynamicRecord(
-			tlv.LockTimeOnionType, &h.OutgoingTimeLock, func() uint64 {
-				return tlv.SizeTUint32(h.OutgoingTimeLock)
-			},
-			tlv.ETUint32, tlv.DTUint32,
-		),
-		tlv.MakePrimitiveRecord(tlv.NextHopOnionType, &nextChanID),
+		record.NewAmtToFwdRecord(&amt),
+		record.NewLockTimeRecord(&h.OutgoingTimeLock),
+		record.NewNextHopIDRecord(&nextChanID),
```

Comment:
With regards to layering: there is a difference in the specification of the route (`route.Route`) and how to serialize this on the wire. I think ideally those things should stay separate. Maybe `PackHopPayload` and other wire-specific functions could all be moved into `record`?

Reply:
interesting point, probably not a bad idea. tho primary purpose of `record` atm is just to reduce code duplication

Reply:
I don't think it is a must for now. But if forces in that direction show up anyway, it may be good to keep in mind.

Reply:
Definitely, fortunately I think it’s the direction we’re headed 👍